### ISSUE TITLE
Regression: dict.get now returns Any, instead of Any | None

### ISSUE THREAD

**Issue by last-partizan** [microsoft/pyright#5733]
**Describe the bug**
I'm not sure this is desired change or regression, but please take a look.

In `pyright@1.1.322` this expression was detected as `"Any | None"`, but new `pyright@1.1.323` detects it as `Any`.

Probably, `Any` includes `None` in principle, but previous variant was more useful.

**Code or Screenshots**

```python
from typing import Any

d: dict[str, Any] = {}
v = d.get("k", None)

reveal_type(v)
```

```
 ~ > npx pyright@1.1.322 /tmp/test.py
/tmp/test.py
  /tmp/test.py:11:13 - information: Type of "v" is "Any | None"
0 errors, 0 warnings, 1 information
 ~ > npx pyright@1.1.323 /tmp/test.py
/tmp/test.py
  /tmp/test.py:11:13 - information: Type of "v" is "Any"
```

**Comment by erictraut** [comment]
This is due to a recent change in typeshed. Pyright 1.1.323 includes the latest typeshed stubs including this change.

If you have concerns about this change, please bring it up with the maintainers of the typeshed stubs.
